### PR TITLE
RTOS lora 

### PR DESCRIPTION
**Pull request by Michal-Brzezinski**

### CODE REVIEW

File: main/include/lora.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include <HardwareSerial.h>
+
+
+namespace lora
+{
+    extern HardwareSerial LoRaWioE5;
+    inline String log_message;
+
+    void init();
+    void send(String wiadomosc);
```
Comment (Serferrr) [inline main/include/lora.h]: change name to "message"

File: main/src/lora.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#include "lora.h"
+#define TX 15
```
Comment (Serferrr) [inline main/src/lora.cpp]: change defines to constexpr (see bmp.cpp file on how to do it)

File: main/src/bmp.cpp [hunk 3]
```diff
@@ -17,7 +17,7 @@ Adafruit_BMP280 bmp_obj;
 void init() {
   if (!bmp_obj.begin(CHIP_ADDR, CHIP_BME)) {
     Serial.println("Viable sensor BMP280 not found, check wiring!");
-    while (1);
+    //while (1);
```
Comment (Serferrr) [inline main/src/bmp.cpp]: why would you comment it out?

File: main/src/lora.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  // setup LoRa for transmitting
+  LoRaWioE5.print(
+      "AT+TEST=RFCFG, 868, SF12, 125, 12, 15, 14, ON, OFF, OFF\r\n");
+  delay(1000);
+}
+
+String gen_rand() {
+  String result = "";
+  srand(time(0));  // inicjalizacja generatora liczb losowych
```
Comment (Serferrr) [inline main/src/lora.cpp]: Some comments are in Polish, some in English and most of them are not needed, please fix

**Review by Serferrr — COMMENTED** [review]
Overall seems good, just some small fixes needed

File: main/src/bmp.cpp (hunk 3 above)
Comment (Bartosz-Slowik) [reply main/src/bmp.cpp]: It was for testing purposes as this line would freeze peogram, we maybe need to think of a solution for testing with missing modules

File: main/include/lora.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    String gen_rand();
+    void lora_log(void* pvParameters);
+
+
+} // namespace Lora
```
Comment (Bartosz-Slowik) [inline main/include/lora.h]: // namespace lora

File: main/platformio.ini [hunk 6]
```diff
@@ -8,11 +8,18 @@
...
+board = lolin32_lite
 framework = arduino
-monitor_speed = 115200
+monitor_speed=115200
 
 lib_deps = mikalhart/TinyGPSPlus@^1.1.0
-           adafruit/Adafruit BMP280 Library@^2.6.8
\ No newline at end of file
+           adafruit/Adafruit BMP280 Library@^2.6.8
+
```
Comment (Bartosz-Slowik) [inline main/platformio.ini]: Could you maybe try to uncomment everything here - I mean keep two configurations, and try if you can run the program?

Platformio might be able to detect the correct one by itself

File: main/src/lora.cpp [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  // communication with LoRa Wio E5, RX and TX pins, baudrate 9600
+  LoRaWioE5.begin(9600, SERIAL_8N1, TX, RX);
+  LoRaWioE5.print("AT+MODE=TEST\r\n");  // set transmitter in test mode
+  delay(1000);
+
+  // check if lora is available
+  if (LoRaWioE5.readString() != "+MODE: TEST") {
+    Serial.println("\nLora is not available");
+  }
+  delay(1000);
```
Comment (Bartosz-Slowik) [inline main/src/lora.cpp]: Remove

File: main/src/lora.cpp [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+void lora_log(void* pvParameters){
+  for(;;){
+    String send_message = log_message;
+    send(send_message);
+    vTaskDelay(pdMS_TO_TICKS(500));
+  }
+
+}
+
+}  // namespace Lora
```
Comment (Bartosz-Slowik) [inline main/src/lora.cpp]: lora

File: main/src/main.cpp [hunk 9]
```diff
@@ -36,16 +39,21 @@ void setup() {
...
   xTaskCreate(gps::gps_task, "gps", 10000, NULL, 1, NULL);
   xTaskCreate(log, "log", 10000, NULL, 1, NULL);
   xTaskCreate(bmp::get_bmp, "bmp", DEFAULT_TASK_SIZE, NULL, 1, NULL);
   xTaskCreate(bmp::print_data, "bmp print", DEFAULT_TASK_SIZE, NULL, 1, NULL);
+  xTaskCreate(lora::lora_log, "lora", DEFAULT_TASK_SIZE, NULL, 1, NULL);
 
   memory::print_data();
   memory::config = memory::Config{222, 456.78, "Hello, EEPROM2!"};
   memory::save_config(memory::config);
+
```
Comment (Bartosz-Slowik) [inline main/src/main.cpp]: Don't andd unnecessary whitespace/empty lines

**Review by Bartosz-Slowik — CHANGES_REQUESTED** [review]

File: main/platformio.ini (hunk 6 above)
Comment (Serferrr) [reply main/platformio.ini]: It can't detect it automatically as far as I tested it but you can just manually set the config once in platformio GUI (at the bottom panel) and it will build only this one config

**Comment by Bartosz-Slowik** [comment]
[like]  Bartosz Słowik reacted to your message:
________________________________

File: main/src/bmp.cpp (hunk 3 above)
Comment (regeliv) [reply main/src/bmp.cpp]: #22

**Review by Bartosz-Slowik — APPROVED** [review]
lgtm
